src/bundles/ipfs-provider.js

```diff
@@ -150,14 +158,20 @@ const readSetting = (id) => {
     }
 
     try {
-      return JSON.parse(setting)
+      return JSON.parse(setting || '')
```

Comment:
Because setting could be a `null` (e.g. if try above failed), while `JSON.parse` has type of `(input:string) -> JSON`. Passing `string|null` would be rejected by a type checker. `setting || ''` on the other hand has `string` type.

I would say that error in this scenario is an expected behavior.